Add an EKS disaster recovery guide

ptgott opened the pull request:
When providing disaster recovery guidance for self-hosted Teleport users, it is reasonable to begin with Amazon EKS users, since self-hosted deployments on this platform are well supported. Provide a guide for disaster recovery on EKS in the case of a regional outage.

This change adds the guide to the `helm-deployments` section, which contains other guidance for self-hosted Teleport clusters on Kubernetes. It changes the section index page to use `DocCardList` instead of an incomplete, manually maintained list of guides.

While initial plans for this guide included a `tctl get all` discussion, we don't recommend this approach because it doesn't back up all backend data, so this change does not include this discussion in guide. Instead, add a link to the Backup and Restore guide, which does discuss this command.

Related changes:

- Extract DNS information in the `kubernetes-cluster.mdx` guide into a partial
- Fix an outdated DynamoDB docs link in the Backup and Restore guide

ptgott committed:
Add an EKS disaster recovery guide

When providing disaster recovery guidance for self-hosted Teleport
users, it is reasonable to begin with Amazon EKS users, since
self-hosted deployments on this platform are well supported. Provide a
guide for disaster recovery on EKS in the case of a regional outage.

This change adds the guide to the `helm-deployments` section, which
contains other guidance for self-hosted Teleport clusters on Kubernetes.
It changes the section index page to use `DocCardList` instead of an
incomplete, manually maintained list of guides.

While initial plans for this guide included a `tctl get all` discussion,
we don't recommend this approach because it doesn't back up all backend
data, so this change does not include this discussion in guide. Instead,
add a link to the Backup and Restore guide, which does discuss this
command.

Related changes:

- Extract DNS information in the `kubernetes-cluster.mdx` guide into a
  partial
- Fix an outdated DynamoDB docs link in the Backup and Restore guide

ptgott committed:
Respond to pschisa feedback

- Mention the TTL of the Proxy Service DNS records.
- Make the change freeze discussion an optional section under "Guidance"
  instead of part of the main procedure, since shutting down a cluster
  makes changes impossible.
- Note how a full regional outage would affect the user's ability to
  execute the "Stop the existing cluster" step.
- Use `helm uninstall` instead of deleting deployments in order to stop
  the existing cluster.
- Add a note about recovery point objective.

ptgott commented on docs/pages/admin-guides/deploy-a-cluster/helm-deployments/disaster-recovery.mdx:
```diff
@@ -0,0 +1,338 @@
+1. Teleport Agents can remain connected to a Teleport cluster even after an
+   outage has caused it to become unavailable. For any Agents running outside
+   the region affected by the disaster, plan to restart them once you have
+   changed the DNS record for your Teleport cluster to terminate any lingering
```
I added this step since @hugoShaka noted that agents can maintain connections to a cluster that has sustained an outage. Would there be any other steps we could take to ensure that no agents are connected to the old cluster?

hugoShaka replied:
You need to either:
- restart the agent
- restart the proxies in the broken region
- force-close all the traffic in the faulty region

hugoShaka replied:
This is something we are trying to fix in the product, but currently the agent can just stick to a broken proxy and not reconnect to the working region, even after you moved the broken region out of the DNS pool.

ptgott replied:
What do we mean by force-closing traffic in the region in this case? Is there a way to do this without restarting the Teleport Agent process? Would the admin need to be able to, say, close connection sockets manually on the underlying node?

Is there a log line that an admin could detect on an affected agent related to connecting to an otherwise unavailable Proxy Service instance? If so, I could indicate that agent failover _usually_ requires no restart, but you should still watch for any agents that have stuck connections and restart those manually.

ptgott replied:
I've suggested a change in f702b371d2d6be1eaec102504746c78a5330644b

ptgott committed:
Respond to hugoShaka feedback

Describe restarting the Proxy Service in relation to gRPC connections as
a way to prevent agents from remaining connected to affected Proxy
Service instances.
